### PR TITLE
Rails migrations to support experiments

### PR DESCRIPTION
**Pull request by balderdash**
Splitting out migrations to merge first.

### CODE REVIEW

File: dashboard/db/migrate/20170331235243_create_experiments.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class CreateExperiments < ActiveRecord::Migration[5.0]
+  def change
+    create_table :experiments do |t|
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170331235243_create_experiments.rb:3]: Do we want to make the timestamp columns all in `at`? E.g., `start_at`, `end_at`, `earliest_section_start_at`, `latest_section_start_at`?

File: dashboard/db/migrate/20170403224005_add_first_activity_time_to_section.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class AddFirstActivityTimeToSection < ActiveRecord::Migration[5.0]
+  def change
+    # Give existing rows a fake first_activity_at that's in the past, but
+    # newly created rows should default to nil
+    add_column :sections, :first_activity_at, :datetime, default: DateTime.parse('1st Jan 1970')
+    change_column_default :sections, :first_activity_at, from: DateTime.parse('1st Jan 1970'), to: nil
```
Comment (ashercodeorg) [inline dashboard/db/migrate/20170403224005_add_first_activity_time_to_section.rb]: Is there a compelling reason not to just start by defaulting the column value to `nil`? And, as necessary, back-populate the values quickly after the migration?

File: dashboard/db/migrate/20170331235243_create_experiments.rb (hunk 1 above)
Comment (balderdash) [reply dashboard/db/migrate/20170331235243_create_experiments.rb:3]: Done, I wasn't sure if that convention applied to all timestamp fields.

File: dashboard/db/migrate/20170403224005_add_first_activity_time_to_section.rb (hunk 2 above)
Comment (balderdash) [reply dashboard/db/migrate/20170403224005_add_first_activity_time_to_section.rb]: No, and it's probably better to have a deadline to finish the backfill anyway.

**Pull request merged**